src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1850,33 +1850,15 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
             if (read < sizeof(buffer))
             {
+                // Normally the cursor should already be at the start of the line, but let's be absolutely sure it is.
+                if (_terminal->GetCursorPosition().x != 0)
+                {
+                    _terminal->Write(L"\r\n");
+                }
+                _terminal->Write(message);

```

Comment:
Moving it into the loop allows us to reuse the console lock that we've already acquired.